espflash/src/target/mod.rs

```diff
@@ -416,19 +416,104 @@ impl Chip {
     /// Given an active connection, read the specified field of the eFuse
     /// region.
     #[cfg(feature = "serialport")]
+    #[deprecated(note = "This only support u32. Use read_efuse_le instead.")]
     pub fn read_efuse(&self, connection: &mut Connection, field: EfuseField) -> Result<u32, Error> {
-        let mask = if field.bit_count == 32 {
-            u32::MAX
-        } else {
-            (1u32 << field.bit_count) - 1
+        if field.bit_count > 32 {
+            return Err(Error::EfuseFieldTooLarge);
+        }
+
+        self.read_efuse_le::<u32>(connection, field)
+    }
+
+    /// Given an active connection, read the specified field of the eFuse
+    /// in little endian order.
+    #[cfg(feature = "serialport")]
+    pub fn read_efuse_le<T: bytemuck::AnyBitPattern>(
+        &self,
+        connection: &mut Connection,
+        field: EfuseField,
+    ) -> Result<T, Error> {
+        // this is a port of the corresponding function in esp-hal
+        // see <https://github.com/esp-rs/esp-hal/blob/b992e944a61b19f6096af5958fd88bb05b02eec2/esp-hal/src/efuse/mod.rs#L119-L183>
+        let EfuseField {
+            block,
+            bit_start,
+            bit_count,
+            ..
+        } = field;
+
+        fn block_address(chip: &Chip, block: u32) -> u32 {
+            let block0_addr = chip.efuse_reg() + chip.block0_offset();
+
+            let mut block_offset = 0;
+            for b in 0..block {
+                block_offset += chip.block_size(b as usize);
+            }
+
+            block0_addr + block_offset
+        }
```

Comment:
Since this is the same logic as `read_efuse_raw`, any reason for not making this a method so the logic is shared between the two methods?